src/app/components/Citations/APACitation.jsx

```diff
@@ -0,0 +1,170 @@
+    this.linkStr = this.props.link_data.link ? <a href={`https://${this.props.link_data.link}`}>{this.props.link_data.link}</a> : '';
+    this.volumeStr = ` (${this.props.version_data.volume})`;
+  }
+
+  returnFormattedCitation() {
+    this.getComponentStrs();
+    if (this.props.government_report === 1) {
+      return this.returnGovernmentReport();
+    }
+
+    if (this.props.version_data.volume) {
+      return this.returnMonographCitation(true);
+    }
+
+    return this.returnMonographCitation();
+  }
+
+  returnGovernmentReport() {
+    let responsibilityStr = '';
+    if (this.props.agents.authors.length > 0) {
+      responsibilityStr = this.authorsStr;
+    } else if (this.props.agents.editors.length > 0) {
+      responsibilityStr = this.editorStr;
+    }
+
+    return (
+      <>
+        {`${responsibilityStr}${this.pubYearStr}. `}
+        <em>{`${this.titleStr} `}</em>
+        {`${this.publisherStr}. `}
+        {this.linkStr}
+      </>
+    );
+  }
+
+  returnMonographCitation(addVolume = false) {
+    let responsibilityStr = '';
+    if (this.props.agents.authors.length > 0) {
+      responsibilityStr = this.authorsStr;
+      if (this.props.agents.editors.length > 0) {
+        this.editionStr = `${this.editionStr} (${this.editorStr}).`;
+      }
+    } else if (this.props.agents.editors.length > 0) {
+      responsibilityStr = this.editorStr;
+    }
+
+    if (this.props.agents.illustrators.length > 0 && this.props.agents.translators.length > 0) {
+      this.titleStr = `${this.titleStr} (${this.illustatorStr}, ${this.translatorStr})`;
+    } else if (this.props.agents.illustrators.length > 0) {
+      this.titleStr = `${this.titleStr} (${this.illustatorStr})`;
+    } else if (this.props.agents.translators.length > 0) {
+      this.titleStr = `${this.titleStr} (${this.translatorStr})`;
+    }
+
+    return (
+      <>
+        {`${responsibilityStr}${this.pubYearStr}. `}
+        <em>{`${this.titleStr} `}</em>
+        {`${addVolume ? this.volumeStr : ''}`}
+        {this.editionStr}
+        {`${this.publisherStr}. `}
+        {this.linkStr}
+      </>
+    );
+  }
+
+  formatAgentNames(agentType, initFirst = false) {
+    const getInitials = name => name.trim().split(/\s+/).map(i => `${i.substr(0, 1).toUpperCase()}.`).join(' ');
+    return this.props.agents[agentType].map((a) => {
+      let initialArray;
+      let lastName;
+      let initials;
+      const cleanName = a.name.replace(/\(.+\)/, '');
+      if (a.name.indexOf(',') > -1) {
+        [lastName, initialArray] = cleanName.split(',');
+        initials = getInitials(initialArray);
+      } else {
+        initialArray = cleanName.split(/\s+/);
+        lastName = initialArray.pop();
+        initials = getInitials(initialArray);
+      }
+      if (initFirst) {
+        return `${initials} ${lastName}`;
+      }
+      return `${lastName}, ${initials}`;
+    }).join(', ');
+  }
+
+  render() {
```

Comment:
This is definitely stylistic, but I am finding this file pretty hard to read.  

I think partly, it's because there's both logic and presentation here, and I think separating it out (see comment in `EditionsList`).  But as a general rule of thumb, since this is mostly a presentation element, I would aim to have most of the markup live here rather than in separate functions.  In situations where you want markup to live in separate functions, I wouldn't nest them more than one level 

(eg: OK, but still better avoided, to go, `<div>{this.getElement()}</div>` but try to avoid `<div>{this.callAnotherFunctionThatReturnsAnElement()}</div>`) 

So here, I think it best to try to have it go:  

```
return (
      <div className="apa-citation body-small">
        <strong>APA</strong>
        <br />
        {isGovernmentReport && returnGovernmentReport()}
        {!isGovernmentReport && returnMonographCitation(isVolume)}  
      </div>
)
```

Reply:
I see what you're saying, I definitely don't disagree and will see what I can do